### PR TITLE
New doc | Preview authentication with subscription selector

### PR DESCRIPTION
**Pull request by dbradish-microsoft**

### CODE REVIEW

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ms.topic: conceptual
+ms.service: azure-cli
+ms.tool: azure-cli
+ms.custom: devx-track-azurecli
+keywords: az login, authentication types, authentication methods, azure, cli login, az login powershell, cli login, sign in, azure cli  
+---
+
+# Choose your subscription when logging into Azure using Azure CLI
+
+With the release of Azure CLI [x.xx.xx](), there is a new login feature that provides a list of available subscriptions. Choose to keep the default subscription, or simply type the line number of the subscription you want to use. Here are the details:
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:16]:
```suggestion
With the preview release of Azure CLI [2.59.0](), there is a new login feature that provides a list of available tenant and subscriptions. Choose to keep the subscription and tenant selected by default, or simply type the line number of the subscription you want to use. Here are the details:
```

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+* The subscription selector is a preview feature available in 64-bit Windows and macOS.
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:18]:
```suggestion
* The subscription selector is available in a preview version of Azure CLI 64-bit Windows and Linux or macOS.
```

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+* The subscription selector is only available when using the `az login` command.
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:19]:
```suggestion
* The subscription selector is only available when using the `az login` command interactively.
```

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+* Azure CLI [x.xx.xx]() must be installed to use the subscription selector. For information on installing the Azure CLI, see [How to install the Azure CLI](./install-azure-cli.md).
+* Installing the preview will over-write your current Azure CLI installation.
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:21]:
```suggestion
* The preview will be installed in the directory you specify and will co-exist with your current Azure CLI installation.
```

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+* You will not be prompted to select a subscription when logging in with a service principal, managed identity or Web Account Manager (WAM).
+
+## Prerequisite for macOS
+
+If you are installing the preview in a macOS environment, Python runtime must be installed.
+
+```bash
+mkdir azure-cli-edge-build 
+cd azure-cli-edge-build 
+git clone https://github.com/jiasli/azure-cli --branch login --depth 1
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:31]:
```suggestion
git clone https://github.com/Azure/azure-cli --branch login --depth 1 
```

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+pip install azdev 
+azdev setup -c 
+az login 
+```
+
+## Install
+
+To install Azure CLI x.xx.x, follow these instructions:
+
+1. Download [Microsoft Azure CLI.zip](https://artprodwus21.artifacts.visualstudio.com/A7b238909-6802-4b65-b90d-184bca47f458/5147fa83-336e-44ef-bbe0-c86b8ae86cbb/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2F6Y2xpdG9vbHMvcHJvamVjdElkLzUxNDdmYTgzLTMzNmUtNDRlZi1iYmUwLWM4NmI4YWU4NmNiYi9idWlsZElkLzE0NTY3MC9hcnRpZmFjdE5hbWUvemlwLXg2NA2/content?format=file&subPath=%2FMicrosoft%20Azure%20CLI.zip ).
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:43]:
```suggestion
1. Download [Microsoft Azure CLI.zip](https://aka.ms/AApydwn).
```

File: docs-ref-conceptual/authenticate-azure-cli.md [hunk 7]
```diff
@@ -24,13 +24,14 @@ There are five authentication options when working with the Azure CLI:
...
 | [Sign in with a managed identity](./authenticate-azure-cli-managed-identity.md) | A common challenge for developers is the management of secrets, credentials, certificates, and keys used to secure communication between services. Using a managed identity eliminates the need for you to manage these credentials.
 | [Sign in using Web Account Manager (WAM)](./authenticate-azure-cli-web-account-manager.md) | WAM is a Windows 10+ component that acts as an authentication broker. WAM provides enhanced security and enhancements are shipped with Windows.
 
 ## Find or change your current subscription
 
-After you sign in, CLI commands are run against your default subscription. If you have multiple subscriptions, you can change your default subscription using `az account set --subscription`. 
+After you sign in, CLI commands are run against your default subscription. If you have multiple subscriptions, you can change your default subscription using `az account set --subscription`. Another option is to upgrade to Azure CLI [preview x.xx.xx]() that provides a subscription list as login and prompts you to accept or select a new default subscription.
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli.md:34]:
```suggestion
After you sign in, CLI commands are run against your default subscription. If you have multiple subscriptions, you can change your default subscription using `az account set --subscription`. Another option is to use the Azure CLI [preview x.xx.xx]() that provides a subscription list as login and prompts you to accept or select a new default subscription.
```

**Review by dcaro — COMMENTED** [review]
Suggested few changes.

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+ms.topic: conceptual
+ms.service: azure-cli
+ms.tool: azure-cli
+ms.custom: devx-track-azurecli
+keywords: az login, authentication types, authentication methods, azure, cli login, az login powershell, cli login, sign in, azure cli  
+---
+
+# Choose your subscription when logging into Azure using Azure CLI
+
+With the preview release of Azure CLI [2.59.0](./release-notes-azure-cli.md#april-02-2024), there's a new login feature that provides a list of available tenants and subscriptions. Choose to keep the subscription and tenant that is selected by default, or type the line number of the subscription you want to use. Here are the details:
```
Comment (daphnemamsft) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:16]: Maybe you could include that the main benefit of having the list of available tenants and subscriptions is that users dont have to manually set their subscription anymore.

Comment (dbradish-microsoft) [reply docs-ref-conceptual/authenticate-azure-cli-preview.md:16]: @daphnemamsft , good idea. done.

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+az login 
+```
+
+## Install
+
+Install Azure CLI authentication preview by following these steps:
+
+1. Download [Microsoft Azure CLI.zip](https://aka.ms/AApydwn).
+1. Unzip the package to a folder.
+1. Using Cmd.exe or PowerShell, call the Azure CLI by running `<unzipped folder path\>\bin\az.cmd`.
```
Comment (daphnemamsft) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:47]: In both Powershell and the cmd terminal, when I try to use the folder path for the "Microsoft Azure CLI" folder, I receive an error if I include the folder path with the name as is (including the spaces). The error only goes away if I add dashes to the folder path. I have posted an example below:

Screenshot 2024-04-09 161020

Comment (dbradish-microsoft) [reply docs-ref-conceptual/authenticate-azure-cli-preview.md:47]: Your path contains spaces.  You need double quotes --> cd "c:\myPath\Microsoft Azure CLI\bin"

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+* The subscription selector is available in a preview version of Azure CLI 64-bit Windows, Linux, or macOS.
+* The subscription selector is only available when using the `az login` command [interactively](./authenticate-azure-cli-interactively.md).
+* The authentication preview must be installed to use the subscription selector.
+* The preview is installed in the directory you specify, and coexists with your current Azure CLI installation.
+* The authentication preview doesn't prompt you to select a subscription when you're logging in with a service principal, managed identity, or Web Account Manager (WAM).
+
+With the current practice, the [az account set](/cli/azure/account#az-account-set) command is required to change your subscription. With the subscription selector at time of interactive login, you skip having to [list](/cli/azure/account#az-account-list), [set](/cli/azure/account#az-account-set) and [show](/cli/azure/account#az-account-show) your default subscription.
+
+## Prerequisite for macOS
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:26]: The pre-requisites are for Linux and MacOS

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+cd azure-cli-edge-build 
+git clone https://github.com/Azure/azure-cli --branch login --depth 1 
+python3 -m venv env 
+. env/bin/activate 
+pip install azdev 
+azdev setup -c 
+az login 
+```
+
+## Install
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:41]: Those installation steps are for Windows, they appear to also be for MacOs in the current layout of the page.

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+Install Azure CLI authentication preview by following these steps:
+
+1. Download [Microsoft Azure CLI.zip](https://aka.ms/AApydwn).
+1. Unzip the package to a folder.
+1. Using Cmd.exe or PowerShell, call the Azure CLI by running `<unzipped folder path\>\bin\az.cmd`.
+1. **Close and reopen any active terminal window to use the newly installed preview.**
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:48]: This is not necessary

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+az login 
+```
+
+## Install
+
+Install Azure CLI authentication preview by following these steps:
+
+1. Download [Microsoft Azure CLI.zip](https://aka.ms/AApydwn).
+1. Unzip the package to a folder.
+1. Using Cmd.exe or PowerShell, call the Azure CLI by running `<unzipped folder path\>\bin\az.cmd`.
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:47]: We prefer that the user goes to the "Microsoft Azure CLI\bin" directory and execute the cmd file from there using `& ./az.cmd`

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+1. **Close and reopen any active terminal window to use the newly installed preview.**
+
+> [!NOTE]
+> You must execute Azure CLI commands from the location where you installed the preview, or update your PATH environment variable to include the location. This article takes the approach of changing the working directory.
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:51]: This sentence is correct but contradict what is on line 47.

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+If you installed the authentication preview in `C:\myPath\Azure CLI preview`, your syntax would look like this:
+
+```azurecli
+cd "C:\myPath\Azure CLI preview\bin"
+C:\myPath\Azure CLI preview\bin> ./az.cmd login
+```
+
+## Choose your subscription
+
+1. Note your default subscription indicated with an asterisk (`*`).
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:71]:
```suggestion
1. Note the subscription and tenant currently selected are indicated with an asterisk (`*`).
```

File: docs-ref-conceptual/authenticate-azure-cli-preview.md [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+C:\myPath\Azure CLI preview\bin> ./az.cmd account show
+```
+
+## Check your version
+
+Check the Azure CLI version of the authentication preview using `./az.cmd --version`.
+
+```azurecli
+C:\myPath\Azure CLI preview\bin> ./az.cmd --version
+```
```
Comment (dcaro) [inline docs-ref-conceptual/authenticate-azure-cli-preview.md:93]: I'm not sure this is necessary

**Review by dcaro — CHANGES_REQUESTED** [review]
Please address the comments

**Pull request merged**